src/it/resources/com/google/checkstyle/test/chapter4formatting/rule451wheretobreak/InputLambdaBodyWrap.java

```diff
@@ -50,6 +50,7 @@ private void sumOfSquares() {
   }
 
   private void showComplexNestedLambda() {
+    // this non-compliant as lambda consists of a NOT single expression
     Function<Integer, Function<Integer, Integer>> createAdder =
         x ->
             (y ->
```

Comment:
formatter make it:
```
  private void showComplexNestedLambda() {
    // this non-compliant as lambda consists of a NOT single expression
    Function<Integer, Function<Integer, Integer>> createAdder = x -> (y -> x + y);
  }
```

style guide wording:
>A line is never broken adjacent to the arrow in a lambda, except that a break may come immediately after the arrow if the body of the lambda consists of a single unbraced expression.

terminology:
(parentheses)
[brackets]
{braces}

"single unbraced expression" is not supper clear, if this is applicable to nested. Conceptually wrapped in parentheses is single expression. So this code is gray area.

Reply:
after meditating , I think:
```
    Function<Integer, Function<Integer, Integer>> createAdder =
        x ->
            (y ->
                x + y);
```

is definitely "unbraced expression" and it is single, fact that is parentheses-ed is still definition of single. Because multiple expressions will imply curly braces and `;`.

Reply:
Yep, I agree, it is compliant, and no violation is expected. Please update the comment.

Reply:
updated